dotnet/src/SemanticKernel/KernelBuilder.cs

```diff
@@ -41,11 +41,6 @@ public static IKernel Create()
     /// <returns>Kernel instance</returns>
     public IKernel Build()
     {
-        if (this._httpHandlerFactory != null)
-        {
-            this._config.SetHttpRetryHandlerFactory(this._httpHandlerFactory);
```

Comment:
Shouldn't this be replaced with call to `SetHttpHandlerFactory`

Reply:
That's a good observation, actually we don't use the Kernel private field _httpHandlerFactory in our code, I just removed it.

When we create our Text/Chat completion services we use the `HttpClientProvider` which uses the `KernelConfig` instance to grab the HttpHandler reference.

I just updated the Kernel method `WithHttpHandlerFactory` to call the KernelConfig method: `this._config.SetHttpHandlerFactory(httpHandlerFactory);`, and added Obsolete flag to the existing `WithRetryHandlerFactory`